sensor/kubernetes/clusterstatus/updater.go

```diff
@@ -313,3 +328,59 @@ func NewUpdater(client client.Interface) common.SensorComponent {
 		getProviders: providers.GetMetadata,
 	}
 }
+
+func (u *updaterImpl) getMetadataFromInfrastructureCR(ctx context.Context) (*storage.ProviderMetadata, error) {
+	clusterInfraCR, err := u.client.OpenshiftConfig().ConfigV1().
+		Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return providerMetadataFromInfraCR(clusterInfraCR), nil
+}
+
+func providerMetadataFromInfraCR(infra *configv1.Infrastructure) *storage.ProviderMetadata {
```

Comment:
I feel like this function should probably live somewhere other than `updater.go`. Maybe extract into `openshift.go` or put somewhere in `pkg/cloudproviders`?